borg/fuse.py

```diff
@@ -136,7 +136,13 @@ def process_archive(self, archive, prefix=[]):
                     # Leaf segment?
                     if i == num_segments:
                         if b'source' in item and stat.S_ISREG(item[b'mode']):
-                            inode = self._find_inode(item[b'source'], prefix)
+                            try:
+                                inode = self._find_inode(item[b'source'], prefix)
+                            except KeyError:
+                                file = segment.decode(errors='surrogateescape')
```

Comment:
segment has only the basename, right?
we could also save item[b'path'] into a local var above 132 (or pop() it) and use it here.